cue/cuex/runtime/package.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2022 The KubeVela Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package runtime
+
+import (
+	"cuelang.org/go/cue/build"
+
+	"github.com/kubevela/pkg/apis/cuex/v1alpha1"
+	"github.com/kubevela/pkg/cue/util"
+	"github.com/kubevela/pkg/util/maps"
+)
+
+const (
+	// VelaPrefix default prefix for internal package names
+	VelaPrefix = "vela/"
+)
+
+type internalPackage struct {
+	name     string
+	template string
+	imp      *build.Instance
+	fns      *maps.SyncMap[string, ProviderFn]
+}
+
+func (in *internalPackage) GetProviderFn(do string) ProviderFn {
+	fn, _ := in.fns.Get(do)
+	return fn
+}
+
+func (in *internalPackage) GetName() string {
+	return in.name
+}
+
+func (in *internalPackage) GetPath() string {
+	return VelaPrefix + in.name
+}
+
+func (in *internalPackage) GetTemplates() []string {
+	return []string{in.template}
```

Comment:
Why a string array?

Reply:
Package interface have GetTemplates function to support multiple cue files (as shown in externalPackage). For internalPackage, we do not have multiple template case for now.